src/functions.c

```diff
@@ -465,8 +501,14 @@ void functionListCommand(client *c) {
             }
         }
         dictReleaseIterator(functions_iter);
+
+        if (with_code) {
+            addReplyBulkCString(c, "library_code");
+            addReplyBulkCBuffer(c, li->code, sdslen(li->code));
+        }
     }
     dictReleaseIterator(iter);
+    setDeferredArrayLen(c, len_ptr, reply_len);
```

Comment:
deferring length comes with a price (system calls and packets).
maybe we can use it only in case a pattern is provided, and use pre-determined length when we can.

Reply:
Fixed